### PR TITLE
196 added function to get list of all (accessible) jira projects.

### LINKED ISSUE #196: Projektkonfiguration: Jira Connector holt Liste aller verfügbaren Projekte
**User Story**
Technisches Requirement

**Acceptance Criteria**
- Beim Initialisieren holt der Jira Connector die Liste der Projekte, für die der technische User Zugriff hat.

Potentiell relevante Klassen:
/specmate-jira-connector/src/com/specmate/connectors/jira/internal/services/JiraConnector.java
/specmate-connectors/src/com/specmate/connectors/config/ProjectConfigService.java

### PR DESCRIPTION
**Pull request by hauptmab**
Closes #196

### CODE REVIEW

File: bundles/specmate-jira-connector/src/com/specmate/connectors/jira/internal/services/JiraUtil.java [hunk 1]
```diff
@@ -48,6 +52,28 @@ public static boolean authenticate(String url, String project, String username,
...
+			
+			client = JiraUtil.createJiraRESTClient(serverUrl, username, password);			
+			for (BasicProject project : client.getProjectClient().getAllProjects().claim()) {
+				projectKeys.add( project.getKey() );
+			}
+			
+			return projectKeys;						
+			
+		} catch (Exception e) {
+			throw new SpecmateAuthorizationException("Jira authentication failed.", e);
```
Comment (junkerm) [inline bundles/specmate-jira-connector/src/com/specmate/connectors/jira/internal/services/JiraUtil.java:76]: please add a finally block and close the client

File: bundles/specmate-jira-connector/src/com/specmate/connectors/jira/internal/services/JiraUtil.java [hunk 2]
```diff
@@ -48,6 +52,28 @@ public static boolean authenticate(String url, String project, String username,
...
+			List<String> projectKeys = new ArrayList<>();
+			
+			client = JiraUtil.createJiraRESTClient(serverUrl, username, password);			
+			for (BasicProject project : client.getProjectClient().getAllProjects().claim()) {
+				projectKeys.add( project.getKey() );
+			}
+			
+			return projectKeys;						
+			
+		} catch (Exception e) {
```
Comment (junkerm) [inline bundles/specmate-jira-connector/src/com/specmate/connectors/jira/internal/services/JiraUtil.java]: please be more specific which exception you catch and throw (you can also use SpecmateInternalException with a specified error code

File: bundles/specmate-jira-connector/src/com/specmate/connectors/jira/internal/services/JiraUtil.java [hunk 3]
```diff
@@ -48,6 +52,28 @@ public static boolean authenticate(String url, String project, String username,
 		}
 		return true;
 	}
+	
+	/**
+	 * Gets a list of all project-keys a given pair of credentials has access to.
+	 */
+	public static List<String> getProjects(String serverUrl, String username, String password)
```
Comment (junkerm) [inline bundles/specmate-jira-connector/src/com/specmate/connectors/jira/internal/services/JiraUtil.java:60]: please use auto format

**Review by junkerm — CHANGES_REQUESTED** [review]

File: bundles/specmate-jira-connector/src/com/specmate/connectors/jira/internal/services/JiraUtil.java (hunk 1 above)
Comment (hauptmab) [reply bundles/specmate-jira-connector/src/com/specmate/connectors/jira/internal/services/JiraUtil.java:76]: improved exception handling

File: bundles/specmate-jira-connector/src/com/specmate/connectors/jira/internal/services/JiraUtil.java (hunk 2 above)
Comment (hauptmab) [reply bundles/specmate-jira-connector/src/com/specmate/connectors/jira/internal/services/JiraUtil.java]: changed exception handling types

File: bundles/specmate-jira-connector/src/com/specmate/connectors/jira/internal/services/JiraUtil.java (hunk 3 above)
Comment (hauptmab) [reply bundles/specmate-jira-connector/src/com/specmate/connectors/jira/internal/services/JiraUtil.java:60]: done

**Review by junkerm — APPROVED** [review]

**Comment by hauptmab** [comment]
This pull request has been part of using https://github.com/qualicen/specmate/pull/236

**Pull request closed without merging**